### DISCUSSION TITLE
Accept `null` as a valid geo field

### DISCUSSION THREAD

**Discussion by irevoire** [meilisearch/product#586] [Feedback & Feature Proposal]
I just realized that in a lot of strongly typed language, you're going to send all the fields of your object.
Thus, if some of your documents have a geo field and others don't (which is really common), there is good luck you're going to send a `"_geo": null,` in your document.

Maybe we should accept that value as a non-set geo field instead of throwing an error?
Also, it's non-breaking, thus it can be done after the v1.0.

**Comment by curquiza** [comment]
I agree, I don't consider this change as breaking personally
- Users with `geo: null` in their datasets could finally send their data without modifying it. If for any reason, they had a condition in their code to catch the error and adapt the data sent, no problem for them either, because they will not go into the condition anymore.
- Users with no `geo: null` in their dataset will not be impacted.

**Comment by gmourier** [comment]
I agree with both of you, we should make that change. Can we schedule it for `v1.1` @curquiza? The amount of work seems to be relatively short (I've quickly discussed it with @irevoire)

Thanks!

**Reply by curquiza** [reply]
If it's quick and @irevoire is available (after addition/deletion batching + "did you mean" error improvement + CSV delimiter), I agree with this change
I need an issue to be opened 😇

**Reply by irevoire** [reply]
Here’s the issue; https://github.com/meilisearch/meilisearch/issues/3497

**Comment by gmourier** [comment]
Hello, everyone following this discussion 👋 

We have just released the first RC (release candidate) of Meilisearch containing this new implementation!

You can test it by using
-  the [release assets](https://github.com/meilisearch/meilisearch/releases)
- the [Meilisearch Docker image](https://hub.docker.com/repository/docker/getmeili/meilisearch/general)

```bash
docker run -it --rm -p 7700:7700 -v $(pwd)/meili_data:/meili_data getmeili/meilisearch:v1.1.0-rc.0
```

You are welcome to communicate any feedback about this new implementation in this discussion.

If you encounter any bugs, please report them [here](https://github.com/meilisearch/meilisearch/issues/new?assignees=&labels=&template=bug_report.md&title=).
Thanks in advance for your help and your involvement in Meilisearch ❤️ 

🎉 Official and stable release containing this change will be available on 3rd April 2023

⚠️ RC (release candidates) are not recommended for production

**Comment by gmourier** [comment]
Hey folks 👋

[v1.1](https://github.com/meilisearch/meilisearch/releases/tag/v1.1.0) has been released. Meilisearch now support the `null` value for the `_geo` field ✨
